### PR TITLE
Add Bunnyhopping

### PR DESCRIPTION
**Pull request by omnit3a**
This PR adds an easily adjustable bunnyhop to the movement code. The length and apex of the jump may easily be modified.
Some tweaking may be necessary.

### CODE REVIEW

**Comment by PQCraft** [comment]
Looks good :+1: 
Though I will need to edit the style a bit and I will have to move or remove the code once AABB collision is added

**Comment by omnit3a** [comment]
ok. I think once (if ever) it gets committed, we should probably make the jumps not as long. I tried dividing the length by `1.5`, which feels more natural to me than the current `1.2` value

File: src/game/game.c [hunk 1]
```diff
@@ -475,7 +476,13 @@ bool doGame(char* addr, int port) {
         if (!onblock) {
             if (yvel > -7.5) {
                 yvel -= 0.5 * pmult;
-            }
+		if (yvel < 2){							//code for bunnyhopping; this line makes sure bunny hopping only works when jumping upwards
+		    float txcm = (sqrt((xcm*xcm) + (zcm*zcm)) / 2);		//these two near identical lines get the magnitude of each direction
+		    float tzcm = (sqrt((xcm*xcm) + (zcm*zcm)) / 2);			
+		    xcm *= txcm / 1.2;						//these two near identical lines apply the magnitude of the directions onto themselves
```
Comment (omnit3a) [inline src/game/game.c:482]: replace 1.2 with 1.5

**Comment by PQCraft** [comment]
Ok will do

**Comment by PQCraft** [comment]
I'll merge this once I get home and have access to git again (~6h)

**Comment by PQCraft** [comment]
Tho after I commit it, it will probably be modified quite a bit lol (in order to match the style for brackets and indentation, and a bit of optimization)

**Pull request merged**